lib/Language/Haskell/Stylish/Parse.hs

```diff
@@ -44,13 +44,12 @@ unCpp = unlines . go False . lines
 
 
 --------------------------------------------------------------------------------
--- | Remove shebang from the first line
+-- | Remove shebang lines
 unShebang :: String -> String
 unShebang str
-    | "#!" `isPrefixOf` str = unlines $ ("" :) $ drop 1 $ lines str
+    | "#!" `isPrefixOf` str = '\n':(unShebang $ unlines $ drop 1 $ lines str)
     | otherwise             = str
```

Comment:
I think this doesn't need to be recursive -- now it keeps using `lines` and `unlines` which seems a bit wasteful.  We can simply use `break` to find the first non-shebang and then replace the lines before that with the empty string:

```
unshebang :: String -> String
unshebang str =
    let (shebangs, other) = break (not . ("#!" `isPrefixOf`)) (lines str) in
    unlines $ map (const "") shebangs ++ other
```